### ISSUE TITLE
Cannot establish repo name from Git remote

### ISSUE THREAD

**Issue by hbldh** [timothycrosley/portray#31]
First off, I want to thank you for this very interesting project!

I was trying to use `portray` for my project, and found that it broke when I tried to run 
```shell script
portray in_browser
```
It said 
```
UserWarning: Unable to identify `repo_name` and `repo_url` automatically
```
I looked through the code and found that this was issued because `portray ` could not determine the name of my project from the git remote. I seldom use `http` for cloning, preferring `ssh` myself and since I also use Azure Devops during working hours I have remotes looking like 
```
username@vs-ssh.visualstudio.com:v3/username/projectname/reponame
```
This `origin`-remote will be ignored by the if-clause in [config.py#L141](https://github.com/timothycrosley/portray/blob/master/portray/config.py#L141).

**Comment by sindrehan** [comment]
I ran into these warnings myself, but as far as I can tell the values `repo_name` and `repo_url` aren't actually used for anything? And there doesn't seem to be a way to set them manually in the pyproject.toml.

**Comment by timothycrosley** [comment]
@hbldh Thank you for this, this is great!

@sindrehan the repo_url is not required, and `portray` does indeed just return a warning and not an error if it can't process it. However, it does enable some cool things if it is provide or correctly auto-determined:

![portray - Your Project with Great Documentation.](https://i.imgur.com/GKzbtv3.png)]

The repo_url is a config option that comes from mkdocs and is used by many mkdocs themes: [https://www.mkdocs.org/user-guide/configuration/#repo_url](https://www.mkdocs.org/user-guide/configuration/#repo_url). So the way to manually specify it in portray would be to specify the mkdocs sub-settings within your `pyproject.toml`:

```toml
[tool.portray.mkdocs]
repo_url = "my_repo_url"
repo_name = "my_repo_name"
```
I'll try to add this to the portray documentation soon. 

Thanks!

~Timothy

**Comment by hbldh** [comment]
@timothycrosley Will the repo urls from ssh remotes have any meaning for mkdocs or did this change merely remove a warning from portray? Will the manually defined repo urls override the ones from this auto detect?

**Comment by sindrehan** [comment]
@timothycrosley Ah, I see, thanks! I just did a quick grep over the repo for `repo_url` and `repo_name` and could only find references to where the values are set, but since they're passed to mkdocs that makes sense. 

Also, I think it would make sense to not raise a warning if the user has specified the url/name manually.

**Comment by timothycrosley** [comment]
I've included repo discovery improvements and additional documentation in the [just released 1.3.0 version of portray](https://timothycrosley.github.io/portray/CHANGELOG/#130-september-15-2019).

I've added a guide on how to deal with the warning and what it means in the [updated troubleshooting guide](https://timothycrosley.github.io/portray/TROUBLESHOOTING/#unable-to-identify-repo_name-repo_url-and-edit_uri-automatically

@hbldh It turns out I'm able to often determine the http link from a the ssh URL making it useful. User configs will indeed take precedence over and auto determinations.

@sindrehan Happy to help! Agreed! I've made this change in the lastest release, empowering users to remove this warning.

Thanks!

~Timothy
